rbac/management/group/view.py

```diff
@@ -475,3 +478,34 @@ def roles(self, request, uuid=None):
             return Response(status=status.HTTP_204_NO_CONTENT)
 
         return Response(status=status.HTTP_200_OK, data=response_data.data)
+
+    def obtain_roles(self, request, group):
+        """Obtain roles based on request, supports exclusion."""
+        exclude = self.validate_and_get_exclude_key(request.query_params)
+
+        roles = (group.roles_with_access() if exclude == 'false'
+                 else self.obtain_roles_with_exclusion(request, group))
+
+        return [RoleMinimumSerializer(role).data for role in roles]
+
+    def obtain_roles_with_exclusion(self, request, group):
+        """Obtain the queryset for roles based on scope."""
+        scope = request.query_params.get('scope', 'account')
+        # Get roles in principal or account scope
+        roles = (get_object_principal_queryset(request, scope, Role) if scope != 'account'
```

Comment:
Just a nit, but it might read easier if we could do `if scope == 'principal'` here?